circuits/circom/trees/incrementalMerkleTree.circom

```diff
@@ -104,10 +106,93 @@ template CheckRoot(levels) {
     // Initialize hashers for intermediate levels, each taking the outputs of two hashers from the previous level.
     var k = 0;
     for (var i = numLeafHashers; i < numLeafHashers + numIntermediateHashers; i++) {
-        computedLevelHashers[i] = PoseidonHasher(2)([hashers[k*2], hashers[k*2+1]]);
+        computedLevelHashers[i] = PoseidonHasher(2)([computedLevelHashers[k*2], computedLevelHashers[k*2+1]]);
         k++;
     }
 
     // Connect the output of the final hasher in the array to the root output signal.
     root <== computedLevelHashers[numHashers-1];
-}
\ No newline at end of file
+}
+
+/**
+ * Calculates the path indices required for Merkle proof verifications. 
+ * Given a node index within an IMT and the total tree levels, it outputs the path indices leading to that node.
+ * The template handles the modulo and division operations to break down the tree index into its constituent path indices.
+ */
+template MerkleGeneratePathIndices(levels) {
```

Comment:
Maybe let's move it to a separate file?

Reply:
I think we could keep it inside this file - We have all binary tree templates in this, and all quinary tree templates in `incrementalQuinTree`.  But perhaps you had a more detailed structure in mind? open to moving stuff around if we apply it across all other imt files

Reply:
It's just because of code style like: one file for one class (or in our case template).

Reply:
What you think about then having two folders inside `circuits/circom/trees` one for `quinTree` and one for `binaryTree` and inside these split the templates one per file?

Reply:
Yes, it works!

Reply:
opened issue here #1419